src/features/tracks/hooks/use-search-params-state.tsx

```diff
@@ -0,0 +1,56 @@
+import { getRouteApi } from '@tanstack/react-router';
+import { PaginationState, SortingState } from '@tanstack/react-table';
+import { isValidSortValue } from '../lib/type-guards';
+
+export const useSearchParamsState = () => {
+  const route = getRouteApi('/tracks');
+  const { page, size, sort, order, q } = route.useSearch();
+  const navigate = route.useNavigate();
+
+  const paginationState = { pageIndex: page - 1, pageSize: size };
+  const sortingState = sort ? [{ id: sort, desc: order === 'desc' }] : [];
+
+  const updatePagination = (pagination: PaginationState) => {
+    void navigate({
+      search: prev => ({
+        ...prev,
+        page: pagination.pageIndex + 1,
+        size: pagination.pageSize,
+      }),
+    });
+  };
+
+  const updateSorting = (sorting: SortingState) => {
+    const sort = sorting.at(0);
+    if (!sort) {
+      void navigate({
+        search: prev => ({ ...prev, sort: undefined, order: undefined }),
+      });
+    } else {
+      const newSortValue = isValidSortValue(sort.id) ? sort.id : undefined;
+      const newSortOrder = sort.desc ? 'desc' : 'asc';
+      void navigate({
+        search: prev => ({ ...prev, sort: newSortValue, order: newSortOrder }),
+      });
+    }
+  };
+
+  const updateSearch = (searchString: string) => {
+    void navigate({
+      search: prev => ({ ...prev, q: searchString }),
```

Comment:
```suggestion
      search: prev => ({ ...prev, query: searchString }),
```
Avoid one-letter names